test/common/config/api_type_db_test.cc

```diff
@@ -0,0 +1,74 @@
+#include "common/config/api_type_db.h"
+
+// For proto descriptors only
+#include "envoy/config/filter/http/ip_tagging/v2/ip_tagging.pb.h"
```

Comment:
interesting, including header files will include descriptors? only linking isn't enough?

Reply:
This is due to how we are auto-generating the BUILD file deps for APIs based on headers. I will address in the next PR which is dealing with the general solution to type upgrade information.

Reply:
Also, for some proto descriptors, I think those with only service definitions, there are some C++ link issues. This largely shouldn't be an issue for us during API ingestion.